packages/data-transport-layer/src/services/l1-ingestion/service.ts

```diff
@@ -205,7 +206,29 @@ export class L1IngestionService extends BaseService<L1IngestionServiceOptions> {
           await sleep(this.options.pollingInterval)
         }
       } catch (err) {
-        if (!this.running || this.options.dangerouslyCatchAllErrors) {
+        if (err instanceof MissingElementError) {
+          // Different functions for getting the last good element depending on the event type.
+          const handlers = {
+            SequencerBatchAppended: this.state.db.getLatestTransactionBatch,
+            StateBatchAppended: this.state.db.getLatestStateRootBatch,
+            TransactionEnqueued: this.state.db.getLatestEnqueue,
+          }
+
+          // Find the last good element and reset the highest synced L1 block to go back to the
+          // last good element. Will resync other event types too but we have no issues with
+          // syncing the same events more than once.
+          const eventName = err.message.split('missing event: ')[1]
```

Comment:
Upstream changes to the error message could break this. A potential way to decouple this would be:

```
export class MissingElementError extends Error {
  constructor(event: string) {
    super(`missing event: ${event}`)
    this.type = event
  }
}
```

where `type` could be `name` or something and then here there is no need to split on the error message and potentially end up with `undefined`. It feels like having `MissingElementError.type` be an enum `number` that could be used for comparisons would be nice

Reply:
This is a good idea.